### ISSUE TITLE
Drop GCC 4.8 atomic stubs from `caml/camlatomics.h` as GCC 4.9 is required in `configure`

### ISSUE THREAD

**Issue by MisterDA** [ocaml/ocaml#12725]
The configure script reads:
```m4
[AS_CASE([$ocaml_cv_cc_vendor],
    # ...
    [gcc-[[0123]]-*|gcc-4-[[0-8]]],
      # No C11 support
      [AC_MSG_ERROR(m4_normalize([This version of GCC is too old.
        Please use GCC version 4.9 or above.]), 69)],
```
but the `runtime/caml/camlatomics.h` header has compatibility stubs for GCC 4.8:
https://github.com/ocaml/ocaml/blob/efd732249a659b5cd256e75442490a6c68193bf7/runtime/caml/camlatomic.h#L48-L80
Granted, someone could have build OCaml with a 4.9 GCC and then build their code including the headers with a 4.8 GCC. Does that really make sense? If not, I suggest we remove the stubs.
Note that the configure script is correct, [C11 atomics were added in GCC 4.9](https://gcc.gnu.org/gcc-4.9/changes.html). Clang support was likely to be added around 3.3-3.4, between 2012 and 2013.
The atomic stubs were added in https://github.com/ocaml/ocaml/commit/68ebe6633d1a8a7628ccd0387c45fb6f6a02af6e.

**Comment by dra27** [comment]
I think it would be fine to remove the entire block, yes. It's also worth considering C++23 with has `stdatomic.h` - there may be a trajectory where `camlatomic.h` eventually becomes unnecessary.

**Comment by MisterDA** [comment]
Fixed as of #12839.
